runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -183,7 +183,10 @@ public JobSpecification translate(
     String workerHarnessContainerImageURL =
         DataflowRunner.getContainerImageForJob(options.as(DataflowPipelineOptions.class));
     RunnerApi.Environment defaultEnvironmentForDataflow =
-        Environments.createDockerEnvironment(workerHarnessContainerImageURL);
+        Environments.createDockerEnvironment(workerHarnessContainerImageURL)
+            .toBuilder()
+            .addAllCapabilities(Environments.getJavaCapabilities())
```

Comment:
Is there some common utility we should be using here (rather than duplicating the code that's used in all the other portable runners)? What about requirements?

Reply:
Requirements is on the pipeline and that works.

Since Dataflow is creating its own environment it is difficult to have common code because of the differences in the artifact staging and job APIs